support-e2e/tests/genericCheckout.test.ts

```diff
@@ -0,0 +1,130 @@
+import { expect, test } from '@playwright/test';
+import { email, firstName, lastName } from './utils/users';
+import { setTestUserDetails } from './utils/testUserDetails';
+import { checkRecaptcha } from './utils/recaptcha';
+import { fillInCardDetails } from './utils/cardDetails';
+import { fillInDirectDebitDetails } from './utils/directDebitDetails';
+import { fillInPayPalDetails } from './utils/paypal';
+import { setupPage } from './utils/page';
+import { afterEachTasks } from './utils/afterEachTest';
+
+interface TestDetails {
```

Comment:
I would say this is a `type` rather than an `interface`. An `interface` is normally something you implement (extend) and is never really used itself. Similar to a Scala `trait`.

We could actually get rid of this altogether and just use a [`const` assertion](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes-func.html#readonly-and-const) on the `testDetails`.

Reply:
But reading through [the docs](https://www.typescriptlang.org/docs/handbook/typescript-in-5-minutes.html), `interface` does get used a lot.

Reply:
Have read up on `const assertion` and can see it does away with writing the interface explicitly and forces the types to only be those specified in the array const definition. I have reworked to this approach. However, am not sure its particularly readable versus defining the interface (but its only used once in this file). Thanks for pointing out and getting me to understand this concept.